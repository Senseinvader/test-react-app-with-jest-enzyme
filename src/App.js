import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {

    state = {
        counter: 0,
    }

    makeIncrementer = amount => () => {
        this.setState(prevState => ({
           counter: prevState.counter + amount
        }))
    }

    handleIncrement = this.makeIncrementer(1);
    handleDecrement = this.makeIncrementer(-1);

    render() {
        return (
            <div className="App">
                <header className="App-header">
                    <img src={logo} className="App-logo" alt="logo" />
                </header>
                <div>
                    <p>Counter: {this.state.counter}</p>
                    <button onClick={this.handleIncrement} className='increment'>Increment</button>
                    <button onClick={this.handleDecrement} className='decrement'>Decrement</button>
                </div>
            </div>
        );
    }
}

export default App;
