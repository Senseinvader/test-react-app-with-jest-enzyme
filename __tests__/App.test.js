import React from 'react';
import ReactDOM from 'react-dom';
import App from '../src/App';
import './setup/setupTests';

describe('App component', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
        });
    // it('matches the snapshot', () => {
    //    const tree = renderer.create(<App/>).toJSON();
    //    expect(tree).toMatchSnapshot();
    // });
})
