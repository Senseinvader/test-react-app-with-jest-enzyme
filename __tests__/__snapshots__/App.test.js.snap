// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="test-file-stub"
    />
  </header>
  <div>
    <p>
      Counter: 
      0
    </p>
    <button
      className="increment"
      onClick={[Function]}
    >
      +++
    </button>
    <button
      className="decrement"
      onClick={[Function]}
    >
      ---
    </button>
  </div>
</div>
`;
